examples/webhook/README.md

```diff
@@ -0,0 +1,130 @@
+
+The new [Materialize webhook source](https://materialize.com/docs/sql/create-source/webhook) introduces a new way of ingesting data via webhooks. Webhook sources in Materialize expose a public URL that allows other applications to push data into Materialize. This enables real-time data streaming from various sources, like IoT devices or third-party services.
+
+## Creating a Webhooks Source in Materialize
+
+You can create a webhook source in Materialize using the [`CREATE SOURCE`](https://materialize.com/docs/sql/create-source/) SQL statement that specifies details like the source name, cluster, body format, and an optional check for validation.
+
+### Validation
+
+It is common practice to validate webhooks requests to ensure they're legitimate. The `CHECK` clause in Materialize allows you to define a boolean expression that's used to validate each request received by the source.
+
+Here's an example SQL command to create a webhook source with validation:
+
+```sql
+-- Create a shared secret
+CREATE SECRET basic_auth_secret AS 'some-secret-value';
+
+-- Create a cluster
+CREATE CLUSTER my_webhook_cluster SIZE = 'xsmall', REPLICATION FACTOR = 1;
+
+-- Create a webhook source with validation
+CREATE SOURCE my_webhook_source IN CLUSTER my_webhook_cluster FROM WEBHOOK
+    BODY FORMAT JSON
+    CHECK (
+        WITH (
+            HEADERS, BODY AS "request_body",
+            SECRET basic_hook_auth
+        )
+        "headers" -> 'authorization' = "basic_hook_auth"
+    );
+```
+
+> **Note**: Without a `CHECK` statement, all requests will be accepted. To prevent bad actors from inserting data, it is strongly encouraged to define a `CHECK` statement with your webhook sources.
+
+The above example creates a webhook source named `my_webhook_source` in the `my_webhook_cluster` cluster. It uses the `JSON` body format and validates each request using the `CHECK` statement.
+
+The public URL for this webhook source is `https://<HOST>/api/webhook/<database>/<schema>/<src_name>`:
+- `<HOST>` is the hostname of the Materialize cluster.
+- `<database>` is the name of the database where the source is created. Defaults to `materialize`.
+- `<schema>` is the name of the schema where the source is created. Defaults to `public`.
+- `<src_name>` is the name of the source. In this case, it's `my_webhook_source`.
+
+## Simulating IoT Data with `datagen`
+
+In the context of IoT, webhooks are extremely useful.
+
+![](https://imgur.com/EnW33xM.png)
+
+Once you have your Materialize webhook source ready, create a `.env` file with the following variables:
+
+```
+# Webhook
+export WEBHOOK_URL=https://<HOST>/api/webhook/<database>/<schema>/<src_name>
+export WEBHOOK_SECRET=some-secret-value
+```
+
+You can use the [sensors.json](sensors.json) schema:
+
+```bash
+datagen -f webhook -s sensors.json -n 1000
+```
+
+The above command will generate 1000 JSON payloads and send them to your Materialize webhook source, the output should look like this:
+
+```json
```

Comment:
Maybe this code block should not use the json annotation? It’s not valid json so the markdown renders weirdly for me.